### PR TITLE
Fix PushRequest merge

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Please provide a description for what this PR is for.

Skip merge when one is empty.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/model/push_context.go [hunk 1]
```diff
@@ -230,7 +230,7 @@ func (first *PushRequest) Merge(other *PushRequest) *PushRequest {
 	}
 
 	// Merge the target namespaces
-	if len(first.NamespacesUpdated) > 0 || len(other.NamespacesUpdated) > 0 {
+	if len(first.NamespacesUpdated) > 0 && len(other.NamespacesUpdated) > 0 {
```
Comment (rshriram) [inline pilot/pkg/model/push_context.go]: I know I did this change in #17479 in my version. Looking at the logic again, shouldn't we be merging though? the list of namespaces to update, the list of configs to update, etc. ?

Comment (howardjohn) [reply pilot/pkg/model/push_context.go]: I think this is correct actually. What we really want to represent here is `AllNamespaces | ListOfNamespaces`, but go makes this pretty hard to do so we instead model `AllNamespaces` as `nil`. So if either one is nil (all namespaces) we want to send to nil (all namespaces)?

Comment (hzxuzhonghu) [reply pilot/pkg/model/push_context.go]: yes, think of the case a virtualservice triggered push and a service triggered push merge. We expect all namespaces updated.

**Pull request merged**